inject-java-test/src/test/groovy/io/micronaut/inject/visitor/beans/BeanWrapperSpec.groovy

```diff
@@ -25,4 +25,20 @@ class BeanWrapperSpec extends Specification {
         wrapper.getRequiredProperty("name", String) == 'Fred'
         wrapper.getRequiredProperty("age", Integer.class) == 10
     }
+
+    void "test setting a non null with null"() {
+        when:"A wrapper is obtained"
+        def bean = new NullabilityBean()
+        BeanWrapper<NullabilityBean> wrapper = BeanWrapper.getWrapper(bean)
+
+        then:"it is correct"
+        wrapper.bean == bean
+
+        when:
+        wrapper.setProperty("name", null)
+
+        then:
+        noExceptionThrown()
+        bean.name == null
+    }
```

Comment:
I would change this test to assert the opposite of what it is asserting now then annotate it with `@PendingFeature` and explain in the annotation that this behaviour will be changed in 3.0

Reply:
I prefer this because then we have a test in place to ensure the behavior works as it should in 2.x and then the test will fail when the behavior changes. It may be forgotten or missed when the functionality changes if it was marked with `@PendingFeature`. I forgot to add a test for the constructor part, I'll do that

Reply:
That is the whole point of `@PendingFeature` so that you know when the functionality changes it breaks the build hence it is not forgotten or missed :)